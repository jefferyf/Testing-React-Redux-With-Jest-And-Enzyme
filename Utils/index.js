import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from './../src/reducers/index';
import { middlewares } from './../src/createStore';

export const checkProps = (component, expectedProps) => {
  const propsError =  checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsError;
};

export const findByTestAttr = (wrapper, attr) => {
  return wrapper.find(`[data-test='${attr}']`);
};

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initialState);
}