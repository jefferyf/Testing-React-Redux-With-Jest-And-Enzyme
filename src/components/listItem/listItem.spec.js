import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { findByTestAttr, checkProps } from '../../../Utils';
import ListItem from './listItem';

describe('ListItem Component', () => {
  describe('Checking PropTypes', () => {
    it('should not throw a warning', () => {
      const expectedProps = {
        title: 'Title',
        desc: 'Desc'
      };
      const propsError = checkProps(ListItem , expectedProps);
      expect(propsError).toBeUndefined();
    });
  });

  describe('Snapshot Tests', () => {
    it('Should render correctly with title prop', () => {
      const wrapper = shallow(<ListItem />);
      wrapper.setProps({ title: 'Title', desc: 'Description' });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should not render with no title prop', () => {
      const wrapper = shallow(<ListItem />);
      wrapper.setProps({ desc: 'Description' });
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('Should render without error', () => {
    let wrapper,
        props;

    beforeEach(() => {
      props = {
        title: 'Title',
        desc: 'Description'
      };
      wrapper = shallow(<ListItem {...props} />);
    });

    it('Should render a listItem element', () => {
      expect(findByTestAttr(wrapper, 'listItemComponent').length).toBe(1);
    });

    it('Should render a title element', () => {
      expect(findByTestAttr(wrapper, 'title').length).toBe(1);
    });

    it('Should render a description element', () => {
      expect(findByTestAttr(wrapper, 'desc').length).toBe(1);
    });
  });
});