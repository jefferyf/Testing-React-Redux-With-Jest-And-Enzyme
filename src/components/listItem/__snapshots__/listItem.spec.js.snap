// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component Snapshot Tests Should not render with no title prop 1`] = `""`;

exports[`ListItem Component Snapshot Tests Should render correctly with title prop 1`] = `
<div
  data-test="listItemComponent"
>
  <h1
    data-test="title"
  >
    Title
  </h1>
  <p
    data-test="desc"
  >
    Description
  </p>
</div>
`;
